File: python/algorithm/everything_algorithm/algorithm_study/추억점수/추억점수.py
def solution(name, yearning, photo):
    answer = []

    table = {}
    for i in range(len(name)):
        table[name[i]] = yearning[i]

    print(table)

    for i in range(len(photo)):
        result = 0
        for j in range(len(photo[i])):
            name = photo[i][j]
            if name in table:
                result += table.get(name)

        answer.append(result)

    return answer

print(solution(["may", "kein", "kain", "radi"],[5, 10, 1, 3], [["may"],["kein", "deny", "may"], ["kon", "coni"]]))
# [5, 15, 0]

# https://school.programmers.co.kr/learn/courses/30/lessons/176963

File: JavaScript/algorithm/src/algorithm/추억점수/추억점수.js
function solution(name, yearning, photo) {
  const answer = [];

  const table = new Map();
  for (let i = 0; i < name.length; ++i) table.set(name[i], yearning[i]);

  for (let i = 0; i < photo.length; ++i) {
    let result = 0;
    for (let j = 0; j < photo[i].length; ++j) {
      const score = table.get(photo[i][j]);
      if (score) result += score;
    }

    answer.push(result);
  }

  return answer;
}

console.log(
  solution(
    ["may", "kein", "kain", "radi"],
    [5, 10, 1, 3],
    [
      ["may", "kein", "kain", "radi"],
      ["may", "kein", "brin", "deny"],
      ["kon", "kain", "may", "coni"],
    ]
  )
);
// [19, 15, 6]

// https://school.programmers.co.kr/learn/courses/30/lessons/176963?language=javascripts
